src/vs/workbench/contrib/browserView/electron-browser/browserEditorInput.ts

```diff
@@ -0,0 +1,232 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Codicon } from '../../../../base/common/codicons.js';
+import { ThemeIcon } from '../../../../base/common/themables.js';
+import { URI } from '../../../../base/common/uri.js';
+import { BrowserViewUri } from '../../../../platform/browserView/common/browserViewUri.js';
+import { EditorInputCapabilities, IEditorSerializer, IUntypedEditorInput } from '../../../common/editor.js';
+import { EditorInput } from '../../../common/editor/editorInput.js';
+import { IThemeService } from '../../../../platform/theme/common/themeService.js';
+import { TAB_ACTIVE_FOREGROUND } from '../../../common/theme.js';
+import { localize } from '../../../../nls.js';
+import { IInstantiationService } from '../../../../platform/instantiation/common/instantiation.js';
+import { IBrowserViewWorkbenchService, IBrowserViewModel } from '../../../services/browserView/common/browserView.js';
+import { hasKey } from '../../../../base/common/types.js';
+import { ILifecycleService, ShutdownReason } from '../../../services/lifecycle/common/lifecycle.js';
+import { BrowserEditor } from './browserEditor.js';
+
+const LOADING_SPINNER_SVG = (color: string | undefined) => `
```

Comment:
I am not sure I have seen this pattern used in other places, where did you get the idea from? Are we missing to support animated codicons in a editor tab?